src/components/EditableHeading/EditableHeading.tsx

```diff
@@ -0,0 +1,168 @@
+import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
+import React, { forwardRef, useLayoutEffect, useMemo, useRef, useState } from "react";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import Heading from "../Heading/Heading";
+import { withStaticProps } from "../../types";
+import styles from "./EditableHeading.module.scss";
+import { HeadingType, HeadingWeight } from "../Heading/HeadingConstants";
+import cx from "classnames";
+import { getStyle } from "../../helpers/typesciptCssModulesHelper";
+import { camelCase } from "lodash-es";
+import HiddenInputPlaceholder from "./HiddenInputPlaceholder";
+import { keyCodes } from "../../constants";
+import useMergeRefs from "../../hooks/useMergeRefs";
+import useClickableProps from "../../hooks/useClickableProps/useClickableProps";
+
+export interface EditableHeadingProps extends VibeComponentProps {
+  value: string;
+  type?: HeadingType;
+  weight?: HeadingWeight;
+  editable?: boolean;
+  onChange?: (value: string) => void;
+  focusOnMount?: boolean;
+  selectOnMount?: boolean;
+  ariaLabel?: string;
+}
+
+const EditableHeading: React.FC<EditableHeadingProps> & {
+  types?: typeof Heading.types;
+  weights?: typeof Heading.weights;
+} = forwardRef(
+  (
+    {
+      id,
+      className,
+      "data-testid": dataTestId,
+      value,
+      type = Heading.types.H1,
+      weight = HeadingWeight.NORMAL,
+      editable = true,
```

Comment:
I usually prefer props that are "off" by default. meaning that if I want to turn something on, I activate it instead of making something inactive

for example, prefer:
`<EditableHeading readOnly/>`
instead of
`<EditableHeading editable={false}/>`